### ISSUE TITLE
Supplying gradients for minimzation has no effect

### ISSUE THREAD

**Issue by JaySandesara** [scikit-hep/iminuit#809]
Hello, I am experiencing the same issue as was discussed here: https://github.com/scikit-hep/iminuit/issues/252 by @HDembinski. 

I have a custom function called NLL. I am trying to perform a fit using iminuit, but noticed that the execution time and performance were sub-optimal. Upon further checking, I found that the following command, where I pass the gradient of the NLL function explicitly to iminuit:

```
m1 = Minuit(jax.jit(NLL), (1.0,0.0), grad=jax.jit(jax.grad(NLL)))
m1.errordef = Minuit.LEAST_SQUARES
m1.tol = 10.0
m1.strategy = 0
m1.migrad()
```

gives the exact same output with the same number of function calls, time of execution and fit values as not providing any gradient at all:

```
m1 = Minuit(jax.jit(NLL), (1.0,0.0), grad=None)
m1.errordef = Minuit.LEAST_SQUARES
m1.tol = 10.0
m1.strategy = 0
m1.migrad()
```

The output for both is given below:

<img width="466" alt="Screenshot 2022-11-03 at 12 45 41 AM" src="https://user-images.githubusercontent.com/52111165/199622898-9dc9e9a0-4d23-4b9e-ab30-699c5678d859.png">

This remains true no matter which strategy I use (0, 1 or 2). I see in https://github.com/scikit-hep/iminuit/issues/252 that the issue was resolved for v1.3. Has this been propagated to v2? I am currently using v2.16.0. Or am I doing something wrong?

  Thanks!

**Comment by HDembinski** [comment]
Gradients work for normal functions where you pass the parameters as individual arguments, see e.g. https://iminuit.readthedocs.io/en/stable/notebooks/automatic_differentiation.html, we also have a unit test for this.

Your code looks ok, the only odd thing is the large value passed to Minuit.tol, but that's probably unrelated. I see that you use the alternative way of passing parameters as a vector. I will check whether that breaks the gradient support.

**Comment by HDembinski** [comment]
I cannot confirm this. The test-case `test_array_func_2` in `tests/test_minuit.py` mimics your case and it calls the gradient function.

If the gradient is used, you should see `Ngrad = Y` next to `Nfcn = X` in the output.

**Comment by HDembinski** [comment]
I am closing this, but feel free to reopen if this is a bug in iminuit after all.
